### ISSUE TITLE
启用保护后无法处理dns解析

### ISSUE THREAD

**Issue by BA7JCM** [AdguardTeam/AdGuardHome#4076]
Have a question or an idea? Please search it [on our forum](https://github.com/AdguardTeam/AdGuardHome/discussions) to make sure it was not yet asked. If you cannot find what you had in mind, please [submit it here](https://github.com/AdguardTeam/AdGuardHome/discussions/new).

### Prerequisites

Please answer the following questions for yourself before submitting an issue. **YOU MAY DELETE THE PREREQUISITES SECTION.**

- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Issue Details

* **Version of AdGuard Home server:**
  * 
  * v0.107.2
* **How did you install AdGuard Home:**
  * 
  * your install scripts
* **How did you setup DNS configuration:**
  * 
  * system
* **If it's a router or IoT, please write device model:**
  * 
* **CPU architecture:**
  * 
  * amd64
* **Operating system and version:**
  * 
  * it install on debian11,my pc operating system is windows 10,my phone operating system is android 11(miui 12.5)

### Expected Behavior

它能正常处理来自局域网的无加密dns和来自广域网的加密dns，同时按照预先设置的过虑列表来去广告

### Actual Behavior

当它开启保护后，我在外面无法使用它进行dns解析，edge浏览器将它设置为doh之后访问网页收到“DNS_PROBE_FINISHED_BAD_SECURE_CONFIG”的提示，手机将它设置为加密dns之后显示“无法连接”，电脑使用nslookup命令行测试时显示“DNS request timed out.”，我把所有过虑列表取消勾选都不行，只有在首页关闭保护才能正常

### Screenshots

<details><summary>Screenshot:</summary>

</details>

**Comment by ainar-g** [comment]
Hello and sorry for the late response.  If I understood correctly, AGH doesn't work when you set it as the DNS server for clients outside of the network?  Is the TLS certificate chain valid for these devices?

**Comment by BA7JCM** [comment]
> Hello and sorry for the late response.  If I understood correctly, AGH doesn't work when you set it as the DNS server for clients outside of the network?  Is the TLS certificate chain valid for these devices?

证书有效，当我将它的保护设置为“启用”的时候，它的DNS功能就停止工作了，不管是来自局域网的无加密DNS还是通过路由器DMZ之后来自广域网的无加密DNS，doh和dot也同样，但是关闭保护后就恢复正常了

**Comment by BA7JCM** [comment]
> Hello and sorry for the late response. If I understood correctly, AGH doesn't work when you set it as the DNS server for clients outside of the network? Is the TLS certificate chain valid for these devices?

The "nslookup" command in the command line under Windows seems to only test unencrypted dns, maybe I don't know how to use it to test encrypted protocol dns

**Comment by BA7JCM** [comment]
> Hello and sorry for the late response. If I understood correctly, AGH doesn't work when you set it as the DNS server for clients outside of the network? Is the TLS certificate chain valid for these devices?

Sometimes after turning on the protection and discovering that something is wrong, you can turn off the protection on the web page, but sometimes it doesn't work, you can only connect to the server through ssh, then directly modify the configuration file, and then restart the adgh service after the change and let It returns to normal, if I operate directly on the web page, there is no response for a long time, and then it returns a "network error" after a while

**Comment by kuke21** [comment]
![5F293689-4878-4C12-8605-42B0708C72E9](https://user-images.githubusercontent.com/55670537/150646381-945ce623-43c5-46ea-8b86-b9d9632c047b.jpeg)
看图解答一切，国内文章都是坑爹的 折磨了3天才看懂 这中文翻译也有水平 不细品真搞不懂意思

**Comment by EugeneOne1** [comment]
@BA7JCM, you may use the [`dnslookup`](https://github.com/ameshkov/dnslookup) utility to test encrypted DNS.  It has binaries for Windows on releases page.

Could you please also collect the [verbose log](https://github.com/AdguardTeam/AdGuardHome/wiki/FAQ#verboselog) for us.  You may either post it here or send to devteam@adguard.com.  Thanks.

**Comment by BA7JCM** [comment]
> @BA7JCM, you may use the [`dnslookup`](https://github.com/ameshkov/dnslookup) utility to test encrypted DNS. It has binaries for Windows on releases page.
> 
> Could you please also collect the [verbose log](https://github.com/AdguardTeam/AdGuardHome/wiki/FAQ#verboselog) for us. You may either post it here or send to [devteam@adguard.com](mailto:devteam@adguard.com). Thanks.

[adgh.log](https://github.com/AdguardTeam/AdGuardHome/files/7943591/adgh.log)

this is log file

**Comment by Birbber** [comment]
@BA7JCM Hi! Sorry for the long silence. Is this issue still relevant?

**Comment by BA7JCM** [comment]
> @BA7JCM Hi! Sorry for the long silence. Is this issue still relevant?

I reinstalled adgh and it doesn't seem like this is happening anymore, but I'm not sure if the problem is fixed
![image](https://user-images.githubusercontent.com/79561519/188264938-60509aea-f7dd-4e5a-8385-a3edb36e99db.png)

In addition, I remember that I unchecked "使用 AdGuard【浏览安全】网页服务" as said https://github.com/AdguardTeam/AdGuardHome/issues/4076#issuecomment-1019299587 and it was normal.

**Comment by BA7JCM** [comment]
> @BA7JCM Hi! Sorry for the long silence. Is this issue still relevant?

In addition, the intercepted domain name still sometimes returns nxdomain, but I chose "default" for "intercept mode"

**Comment by ainar-g** [comment]
If the original issue isn't a problem anymore, I'll close this issue, if you don't mind.

 >  In addition, the intercepted domain name still sometimes returns nxdomain, but I chose "default" for "intercept mode"

I'm not sure what you mean by “intercepted domain”, but please make sure that clients only use AdGuard Home as their DNS servers.  Many OSs will use a secondary resolver if you don't enter the address of AdGuard Home to both primary and secondary DNS IP addresses, for example.

**Comment by BA7JCM** [comment]
> If the original issue isn't a problem anymore, I'll close this issue, if you don't mind.
> 
>  >  In addition, the intercepted domain name still sometimes returns nxdomain, but I chose "default" for "intercept mode"
> 
> I'm not sure what you mean by “intercepted domain”, but please make sure that clients only use AdGuard Home as their DNS servers.  Many OSs will use a secondary resolver if you don't enter the address of AdGuard Home to both primary and secondary DNS IP addresses, for example.

The browser I use is Microsoft edge, I set doh in the browser's privacy settings, and this doh is built by adgh

**Comment by BA7JCM** [comment]
> If the original issue isn't a problem anymore, I'll close this issue, if you don't mind.
> 
>  >  In addition, the intercepted domain name still sometimes returns nxdomain, but I chose "default" for "intercept mode"
> 
> I'm not sure what you mean by “intercepted domain”, but please make sure that clients only use AdGuard Home as their DNS servers.  Many OSs will use a secondary resolver if you don't enter the address of AdGuard Home to both primary and secondary DNS IP addresses, for example.

After setting doh, the browser seems to ignore the DNS settings of the system, and only use the set doh to resolve the domain name

**Comment by BA7JCM** [comment]
> If the original issue isn't a problem anymore, I'll close this issue, if you don't mind.
> 
>  >  In addition, the intercepted domain name still sometimes returns nxdomain, but I chose "default" for "intercept mode"
> 
> I'm not sure what you mean by “intercepted domain”, but please make sure that clients only use AdGuard Home as their DNS servers.  Many OSs will use a secondary resolver if you don't enter the address of AdGuard Home to both primary and secondary DNS IP addresses, for example.

In addition, I want to know if I check "使用 AdGuard【浏览安全】网页服务" and cannot access ADguard's server for some reason (eg the gfw), will it cause adgh to fail to process parsing?

**Comment by BA7JCM** [comment]
> I'm not sure what you mean by “intercepted domain”
I used google translate to translate what I want to say from Chinese to English to reply (including this reply), I mean the domain name blocked by ADguard's DNS filter

**Comment by ainar-g** [comment]
I see.  Please check the Query log.  If the `NXDOMAIN` is there, and the response is marked as coming from the upstream, it is most probably what the upstream server has sent, and so it is not an issue with AdGuard Home.  If there is nothing in the Query log, your browser may still use another client.  Otherwise, please file a new issue.

 >  In addition, I want to know if I check "使用 AdGuard【浏览安全】网页服务" but cannot access ADguard's server for some reason (eg the gfw), will it cause adgh to fail to process parsing?

Yes, the Browsing Security feature uses AdGuard DNS as the source of hashes used to check domain names.  If your instance of AGH cannot reach AdGuard DNS, enabling Browsing Security is likely to cause issues.

(Please use GitHub Discussions for further questions.  Thanks.)
